addons/product_margin/wizard/product_margin.py

```diff
@@ -1,79 +1,42 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
 
 import time
 
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import api, fields, models, _
 
 
-class product_margin(osv.osv_memory):
+class ProductMargin(models.TransientModel):
     _name = 'product.margin'
     _description = 'Product Margin'
-    _columns = {
-        'from_date': fields.date('From'),
-        'to_date': fields.date('To'),
-        'invoice_state': fields.selection([
-            ('paid', 'Paid'),
-            ('open_paid', 'Open and Paid'),
-            ('draft_open_paid', 'Draft, Open and Paid'),
-        ], 'Invoice State', select=True, required=True),
-    }
+    _rec_name = 'from_date'
 
-    _defaults = {
-        'from_date': time.strftime('%Y-01-01'),
-        'to_date': time.strftime('%Y-12-31'),
-        'invoice_state': "open_paid",
-    }
+    from_date = fields.Date(string='From', default=time.strftime('%Y-01-01'))
+    to_date = fields.Date(string='To', default=time.strftime('%Y-12-31'))
+    invoice_state = fields.Selection([
+        ('paid', 'Paid'),
+        ('open_paid', 'Open and Paid'),
+        ('draft_open_paid', 'Draft, Open and Paid'),
+    ], select=True, required=True, default="open_paid")
 
-    def action_open_window(self, cr, uid, ids, context=None):
-        """
-            @param cr: the current row, from the database cursor,
-            @param uid: the current user’s ID for security checks,
-            @param ids: the ID or list of IDs if we want more than one
-
-            @return:
-        """
-        context = dict(context or {})
+    @api.multi
+    def action_open_window(self):
+        context = dict(self.env.context)
 
         def ref(module, xml_id):
-            proxy = self.pool.get('ir.model.data')
-            return proxy.get_object_reference(cr, uid, module, xml_id)
-
-        model, search_view_id = ref('product', 'product_search_form_view')
-        model, graph_view_id = ref('product_margin', 'view_product_margin_graph')
-        model, form_view_id = ref('product_margin', 'view_product_margin_form')
-        model, tree_view_id = ref('product_margin', 'view_product_margin_tree')
+            return self.env['ir.model.data'].xmlid_to_res_id('%s.%s' % (module, xml_id))
 
-        #get the current product.margin object to obtain the values from it
-        records = self.browse(cr, uid, ids, context=context)
-        record = records[0]
+        search_view_id = ref('product', 'product_search_form_view')
+        graph_view_id = ref('product_margin', 'view_product_margin_graph')
+        form_view_id = ref('product_margin', 'view_product_margin_form')
+        tree_view_id = ref('product_margin', 'view_product_margin_tree')
 
-        context.update(invoice_state=record.invoice_state)
+        context.update(invoice_state=self.invoice_state)
```

Comment:
take one dict variable and in last use self.with_context() then pass it in dictionary